### PR TITLE
Simplify internal/strs

### PR DESCRIPTION
**Pull request by amckinney**
As discussed in https://github.com/uber/prototool/pull/50, this makes several changes to the `internal/strs` package in an attempt at simplifying.

A couple functions using the `containsRune` helper (via the `extraRunes ...rune` parameter) were not actually being used, so it seems better to remove for now. The package name linter used this by allowing `.` characters, so we explicitly handle that there.

Given that the [Protobuf spec](https://developers.google.com/protocol-buffers/docs/reference/proto3-spec) strictly specifies valid identifiers as `A-Z`, `a-z` and `_`, this does not use the corresponding `unicode` functions, such as `unicode.IsLetter`. Instead, we explicitly check for the valid identifiers in a few helper functions, `isUpper`, `isLower`, and `isDigit` so that we restrict the valid input to what is expected by Proto. This is functionally the same as before.

### CODE REVIEW

File: internal/x/lint/check_package_lower_snake_case.go [hunk 1]
```diff
@@ -59,7 +61,7 @@ func (v *packageLowerSnakeCaseVisitor) VisitPackage(pkg *proto.Package) {
 
 func (v *packageLowerSnakeCaseVisitor) Finally() error {
 	if v.pkg != nil {
-		if !strs.IsLowerSnakeCase(v.pkg.Name, '.') {
+		if !strs.IsLowerSnakeCase(strings.Replace(v.pkg.Name, ".", "", -1)) {
```
Comment (peter-edge) [inline internal/x/lint/check_package_lower_snake_case.go]: This is super cute, I love it +1

File: internal/strs/strs.go [hunk 2]
```diff
@@ -133,20 +104,10 @@ func ToSnakeCase(s string) string {
 	return output
 }
 
-// DedupeSlice returns s with no duplicates and no empty strings, in the same order.
+// Dedupe returns s with no duplicates and no empty strings, sorted.
```
Comment (peter-edge) [inline internal/strs/strs.go]: nit: I think this should still be `DedupeSort` as a form of self-documentation. If the function is just named `Dedupe`, and I input `[]{"b", "a", "c"}`, my natural instinct would be to expect to get the same slice value back. People may miss the Godoc.

File: internal/strs/strs_test.go [hunk 3]
```diff
@@ -53,8 +52,6 @@ func TestIsLowerSnakeCase(t *testing.T) {
 	assert.False(t, IsLowerSnakeCase("Hello_world"))
 	assert.False(t, IsLowerSnakeCase("_hello_world"))
 	assert.False(t, IsLowerSnakeCase("hello_world_"))
-	assert.False(t, IsLowerSnakeCase("hello_world.foo"))
```
Comment (peter-edge) [inline internal/strs/strs_test.go]: I think this test case should remain.

File: internal/strs/strs_test.go [hunk 4]
```diff
@@ -67,8 +64,6 @@ func TestIsUpperSnakeCase(t *testing.T) {
 	assert.False(t, IsUpperSnakeCase("Hello_world"))
 	assert.False(t, IsUpperSnakeCase("_HELLO_WORLD"))
 	assert.False(t, IsUpperSnakeCase("HELLO_WORLD_"))
-	assert.False(t, IsUpperSnakeCase("HELLO_WORLD.FOO"))
```
Comment (peter-edge) [inline internal/strs/strs_test.go]: I think this test case should remain.

File: internal/strs/strs.go [hunk 5]
```diff
@@ -29,99 +29,70 @@ import (
...
 }
 
-// ToSnakeCase converts s to Snake_case.
+// toSnake converts s to snake_case.
 // It is assumed s has no spaces.
-func ToSnakeCase(s string) string {
+func toSnake(s string) string {
 	output := ""
 	for i, c := range s {
 		if i > 0 && unicode.IsUpper(c) && output[len(output)-1] != '_' && i < len(s)-1 && !unicode.IsUpper(rune(s[i+1])) {
```
Comment (peter-edge) [inline internal/strs/strs.go]: We did `unicode.IsUpper` in #50, we should probably change it back.

File: internal/strs/strs.go [hunk 6]
```diff
@@ -29,99 +29,70 @@ import (
...
 func ToUpperSnakeCase(s string) string {
-	return strings.ToUpper(ToSnakeCase(s))
+	return strings.ToUpper(toSnake(s))
 }
 
-// ToSnakeCase converts s to Snake_case.
+// toSnake converts s to snake_case.
 // It is assumed s has no spaces.
-func ToSnakeCase(s string) string {
+func toSnake(s string) string {
```
Comment (peter-edge) [inline internal/strs/strs.go]: You'll notice in prototool that the general style is to put public functions at the top, and private functions below, so I would move this to the bottom.

File: internal/strs/strs.go (hunk 2 above)
Comment (amckinney) [reply internal/strs/strs.go]: Good call! Will update.

File: internal/strs/strs_test.go (hunk 3 above)
Comment (amckinney) [reply internal/strs/strs_test.go]: Yea good call here - removing it was lazy of me.

File: internal/strs/strs.go (hunk 5 above)
Comment (amckinney) [reply internal/strs/strs.go]: Oops, nice catch! Meant to do that here as well.

File: internal/strs/strs.go (hunk 6 above)
Comment (amckinney) [reply internal/strs/strs.go]: Sweet, works for me.

**Pull request merged**